src/components/api/codenvy-runner.factory.js

```diff
@@ -63,6 +68,33 @@ class CodenvyRunner {
     return promise;
   }
 
+  /**
+   * Gets running processes in workspace
+   */
+  fetchRunningProcesses(workspaceId) {
```

Comment:
Here we can fetch all workspace processes and filter them in controllers by our needs.  
Besides, the variable this.processesPerWorkspaceMap = new Map(); - means all processes, but contains only running.